contrib/cf-remote/cf_remote/remote.py

```diff
@@ -204,3 +219,46 @@ def install_host(
             demo_lib.agent_run(data, connection=connection)
             demo_lib.disable_password_dialog(host)
         demo_lib.agent_run(data, connection=connection)
+
+
+def get_artifact_url(hub, data):
```

Comment:
This function seems a little overkill for deletion, since you could just try the 3 possibilities. Something like:

```
dpkg --purge cfengine3-community || dpkg --purge cfengine3-nova-hub || dpkg --purge cfengine3
```

(Or you could even search for what cfengine is installed).
```

Reply:
Also, I think there is a flaw here. I want to uninstall my hosts like this:

```
cf-remote uninstall --hosts 1.2.3.4,2.2.3.4,3.3.3.4
```

But this will fail if one of those is a hub.